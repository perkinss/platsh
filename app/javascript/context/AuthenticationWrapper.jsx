import React, { useEffect, useState, useContext } from 'react';
import User from '../model/user'

export const RootAuthContext = React.createContext();
export default ({ children }) => {
  let anyAuthVal,anyAuthData;
  try {
    anyAuthVal=JSON.parse(window.sessionStorage.getItem('authenticated'));
  }
  catch(err) {
    anyAuthVal=false;
  }
  try {
    anyAuthData = JSON.parse(window.sessionStorage.getItem('authData'))
  }
  catch(err) {
    anyAuthData = null;
  }
  const guestUser = new User({})
  const prevAuth = (anyAuthVal ? anyAuthVal : false);
  const prevAuthData = (anyAuthData ? anyAuthData : null);
  const prevUser = (anyAuthData ? new User(anyAuthData) : guestUser);
  const [authenticated, setAuthenticated] = useState(prevAuth);
  const [authData, setAuthData] = useState(prevAuthData);
  const [currentUser, setCurrentUser] = useState(prevUser);
  useEffect(
    () => {
      window.sessionStorage.setItem('authenticated', JSON.stringify(authenticated));
      window.sessionStorage.setItem('authData', (authData ? JSON.stringify(authData):null));
      setCurrentUser(authData ? new User(authData) : guestUser)
    },
    [authenticated, authData]
  );
  const defaultContext = {
    authenticated, setAuthenticated,
    authData, setAuthData,
    currentUser, setCurrentUser
  };
  return (
    <RootAuthContext.Provider value={defaultContext}>
      {children}
    </RootAuthContext.Provider>
  );
};
