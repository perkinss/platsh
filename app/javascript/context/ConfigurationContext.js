import React from 'react'

export const configuration = {
    sections: [
        {
            id: null,
            courses: [],
            name: '',
            assessment: [
                {
                    type: 'test',
                    name: '',
                    tasks: [
                    {
                        name: '',
                        available_standards: [
                            {
                                id: -1,
                                level: ''
                            }
                        ],
                        competencies: [

                        ]
                    },
                    ]
                }
            ],
            enrollees: [
                1, 2, 3, 22
            ]
        },
    ],

}


const ConfigurationContext =  React.createContext(configuration);

export { ConfigurationContext }
