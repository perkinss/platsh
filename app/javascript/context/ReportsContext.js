import React from 'react'

export const reports = {
    students: [],
    sections: [],
    selected_section: null,
    selected_student: null,
    section_overview_marks: {},
    section_competency_scores: {},
    student_section_marks: {},
    heat_map_colors: []
}


const ReportsContext =  React.createContext(reports);

export { ReportsContext }
