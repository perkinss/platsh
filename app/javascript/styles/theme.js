import {createMuiTheme} from "@material-ui/core/styles";

const theme = createMuiTheme({
    palette: {
        primary: {
            main: '#512da8',
        },
        secondary: {
            main: '#ab47bc',
        },
        type: 'dark'
    },
    slider: {
        selectionColor: '#F76B1C',
        handleFillColor: '#F76B1C',
        thumb: 'secondary'
    }

})

export default theme
