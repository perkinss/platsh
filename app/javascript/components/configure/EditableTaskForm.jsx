import React, {useContext, useEffect, useState} from 'react'
import {
    Avatar,
    Button,
    Checkbox,
    Chip,
    Collapse,
    Divider,
    Drawer,
    FormControl,
    FormControlLabel,
    Grid,
    Grow,
    List,
    ListItem,
    ListItemIcon,
    ListItemText,
    ListSubheader,
    Paper,
    TextField,
    Tooltip,
    Typography,
    Zoom
} from '@material-ui/core'
import {makeStyles, useTheme} from '@material-ui/styles'
import ExpandMore from '@material-ui/icons/ExpandMoreRounded'
import ExpandLess from '@material-ui/icons/ExpandLessRounded'
import StarBorder from '@material-ui/icons/StarBorder'
import Star from '@material-ui/icons/StarRounded'
import TurnedIn from '@material-ui/icons/TurnedInRounded'
import TurnedInNot from '@material-ui/icons/TurnedInNot'

import {ConfigurationContext} from '../../context/ConfigurationContext'
import Skeleton from "@material-ui/lab/Skeleton";

const makeHeaderStyles =  makeStyles(theme => ({
    root: {
        background: theme.palette.primary.dark,
        opacity: '.99',
        fontSize: '2em',
        width: '100%'
    },

}))

const makeListStyles =  makeStyles(theme => ({
    root: {
        width: '100%'
    }
}))

const drawerWidth = '40%'

const drawerStyles = makeStyles(theme => ({
    drawer: {
        width: drawerWidth,
        flexShrink: 0,
    },
    drawerPaper: {
        width: drawerWidth,
    }
}))

export default function EditableTaskForm(props) {

    let {classes, selectedCourses, possiblyNewTask, editMode, isHolistic, handleDeleteTask, assessmentCompetencyId, loading } = props
    let drawerClasses = drawerStyles()
    const headerclasses = makeHeaderStyles()
    const listClass = makeListStyles()
    const theme = useTheme()
    const {dispatch, state} = useContext(ConfigurationContext)

    const generateSelectedCompetencies = (baseCompetencies) => {
        return baseCompetencies && baseCompetencies.length > 0 ?
            baseCompetencies.reduce((obj, key) => {
                obj[key] = true
                return obj
            }, {}) : {}
    }

    const generateSelectedStandards = (baseStandards) => {
        if (Object.keys(baseStandards).length > 0) {
            return Object.keys(baseStandards).reduce((obj, key) => {
                obj[key] = true
                return obj
            }, {})
        } else {
            return {}
        }
    }

    const [task, setTask] = useState(possiblyNewTask)
    const [taskName, setTaskName] = useState(possiblyNewTask.name)
    const [selectedStandards, setSelectedStandards] = useState(generateSelectedStandards(possiblyNewTask.standards))
    const [selectedCompetencies, setSelectedCompetencies] = useState(generateSelectedCompetencies(possiblyNewTask.competencies))
    const [standards, setStandards] = useState(possiblyNewTask.standards)
    const [competencies, setCompetencies] = useState(possiblyNewTask.competencies)
    const [editing, setEditing] = useState(editMode)


    const [showStandardDrawer, setShowStandardDrawer] = useState(false)
    const [showCompetencyDrawer, setshowCompetencyDrawer] = useState(false)

    const [chipColor, setChipColor] = useState({1: 'secondary'})
    const [expandSublist, setExpandSublist] = React.useState({});

    const handleTaskNameChange = (event) => {
        setTaskName(event.target.value)
    }

    const doDelete = (event) => {
        handleDeleteTask(event, {name: taskName, id: task.id})
        handleDoneTask(event)
    }

    const handleDoneTask = (event) => {
        setEditing(false)
    }

    const handleEdit = (event) => {
        setEditing(!editing)
    }

    const handleOpenStandardsDrawer = () => {
        setShowStandardDrawer(!showStandardDrawer)
    }

    const handleOpenCompetencyDrawer = () => {
        setshowCompetencyDrawer(!showCompetencyDrawer)
    }

    useEffect(() => {
        if (assessmentCompetencyId && competencies.length > 0) {
            let filtered = competencies.filter(id => `${id}` !== `${assessmentCompetencyId}`)
            setCompetencies(filtered)
            setSelectedCompetencies(generateSelectedCompetencies(filtered))
        }

    },[assessmentCompetencyId])

    const standardRows = () => {
        if (state.available_content_standards && standards) {
            let stan = [].concat.apply([],Array.from(state.available_content_standards).map((course) => {
                return [].concat.apply([], course.contents.map((content) => {
                    return content.standards.filter((standard) => {
                        return standards && (Object.keys(standards).includes(`${standard.id}`) || Object.keys(standards).includes(standard.id))
                    })
                }))
            }))
            let chips = stan.map(standard => {
                let chipProps = {}
                if ( editing) {
                    chipProps = { onClick: () => handleClickChip(standard.id) }
                }
                return (
                    <Grow
                        key={`animation-${standard.id}`}
                        in={true}
                        style={{ transformOrigin: '0 0 0' }}
                        timeout={1000}
                    >
                        <div style={{ overflow: "hidden", textOverflow: "ellipsis" }}>
                            <Tooltip title={standard.description} >
                            <Chip
                                key={`standard-${standard.id}`}
                                avatar={
                                    <Avatar>
                                        <Star/>
                                    </Avatar>
                                }
                                label={
                                    <React.Fragment>
                                        <Typography>
                                            {!isHolistic ? <strong> {standards[standard.id]} </strong> : ""}
                                            {standard.description}
                                        </Typography>
                                    </React.Fragment>
                                }
                                className={classes.chip}
                                color={standards[standard.id] === 'M' ? 'secondary' : standards[standard.id] === 'H' ? 'primary' : 'default'}
                                {...chipProps}
                            />
                            </Tooltip>
                        </div>
                    </Grow>
                )
            })

            if (chips.length > 0 ) {
                return (
                    <Grid item sm={12}>
                        { editing && !isHolistic ?
                            <React.Fragment>
                                <Typography variant={'caption'} style={{paddingLeft: '5px', paddingBottom: '5px'}}>
                                Click to set the level of difficulty (<b>L</b>ow, <b>M</b>edium
                                or <b>H</b>igh):</Typography>
                            <br/>
                            </React.Fragment>
                            :
                            <React.Fragment>
                                <Typography variant={'caption'} style={{paddingLeft: '5px', paddingBottom: '5px'}}>
                                    Standards:</Typography>
                                <br/>
                            </React.Fragment>
                        }
                        {chips}
                    </Grid>
                )
            }

        }
    }

    const competencySection = () => {

        if (competencies && competencies.length > 0) {
            let com = [].concat.apply([],Array.from(state.available_grouped_competencies).map((course) => {
                return [].concat.apply([],course.groups.map((group) => {
                    return group.competencies.filter((competency) => {
                        return competencies && (competencies.includes(`${competency.id}`) || competencies.includes(competency.id))
                    })
                }))
            }))

            return (
                <React.Fragment>
                    <Grid item sm={12}>
                        <Divider/>
                        <Typography variant={'caption'}>{ (task && editing) ? 'Selected ' :'' }Competencies: </Typography>
                        <List aria-label={"List of competencies selected for task"} dense={true}>
                            {com.map(competency =>
                                <Zoom in={true} style={{ transitionDelay: '500ms' }}  key={`list-animation-${competency.id}`}>
                                    <ListItem key={`standard-${competency.id}`} >
                                        <ListItemText
                                            className={listClass.root}
                                            disableTypography={true}
                                            primary={
                                                <Tooltip title={competency.description}>
                                                <Typography noWrap={true} variant={'body1'}>{competency.description}</Typography>
                                                </Tooltip>
                                            }
                                            primaryTypographyProps={{noWrap: true}}
                                            />
                                    </ListItem>
                                </Zoom>)}
                        </List>
                    </Grid>
                </React.Fragment>
            )
        }
    }

    const handleClickChip = (value) => {
        if (editing) {
            switch (standards[value]) {
                case 'L' :
                    setStandards({...standards, [value]: 'M'})
                    setChipColor({...chipColor, [value]: 'secondary'})
                    break;
                case 'M' :
                    setStandards({...standards, [value]: 'H'})
                    setChipColor({...chipColor, [value]: 'primary'})
                    break;
                case 'H' :
                    setStandards({...standards, [value]: 'L'})
                    setChipColor({...chipColor, [value]: 'default'})
                    break;
                default:
                    break;
            }
        }

    }

    const handleUpdateTask = (event) => {

        let updatedTask = {id: task.id, name: taskName.trim(), standards: standards, competencies: competencies, deletable: task.deletable }
        setTask(updatedTask)
        dispatch({ type: 'UPDATE_TASK_IN_CURRENT_ASSESSMENT', task: updatedTask })
        if (event.target.innerHTML === 'Done') {
            handleDoneTask(event)
        }
    }

    const handleStandardCheck = (event) => {
        let s = event.target.value
        setSelectedStandards({...selectedStandards, [s]: !selectedStandards[s]})
    }

    const closeDrawer = () => {

        if (showStandardDrawer) {
            const selected = Object.keys(selectedStandards)
                .filter(key => selectedStandards[key]).reduce((obj, key) => {
                    obj[key] = standards[key] || 'M'
                    return obj
                }, {})
            setStandards(selected)
            setShowStandardDrawer(false)
        }

        if (showCompetencyDrawer) {
            const selected = Object.keys(selectedCompetencies)
                .filter(key => selectedCompetencies[key])
            const numbered = selected.map( s => Number(s))
            setCompetencies(selected)
            setshowCompetencyDrawer(false)
        }

    }

    const handleExpandTopic = (index) => {
        if (expandSublist && expandSublist[index]) {
            setExpandSublist({...expandSublist, [index]: false})
        } else {
            setExpandSublist({...expandSublist, [index]: true})
        }
    }

    const handleCompetencyCheck = (event) => {
        let c = event.target.value
        setSelectedCompetencies({...selectedCompetencies, [c]: !selectedCompetencies[c]})
    }

    const getDrawerContents = () => {
        let drawerContent = null

        if (state.available_content_standards.length > 0 && showStandardDrawer) {
            drawerContent = Array.from(state.available_content_standards).map((course, index) => {
                return (
                    <List
                        key={`list-${course.id}-${index}`}
                        component="div"
                        aria-labelledby="nested-list-subheader"

                        subheader={
                            <ListSubheader
                                component="div"
                                inset={true}
                                classes={headerclasses}
                                className={headerclasses.root}
                                styles={{opacity: .25}}
                                id="nested-list-subheader">
                                {course.title}
                            </ListSubheader>
                        }
                        className={classes.root}
                    >
                        {course.contents.map((content, index) => {
                            return (
                                <React.Fragment >
                                    <ListItem key={`list-${index}-${content.id}`} button onClick={() => handleExpandTopic(index)}>
                                        <ListItemText primary={content.name}/>
                                        {expandSublist[index] ? <ExpandLess/> : <ExpandMore/>}
                                    </ListItem>
                                    <div style={{ width: '100%', overflow: "hidden", textOverflow: "ellipsis" }}>
                                    <Collapse in={expandSublist[index]} timeout="auto" unmountOnExit>
                                        {content.standards.map((standard, index) => {
                                            return (
                                                <List key={`l-${index}-${standard.id}`} component="div" disablePadding>
                                                    <ListItem button className={classes.nested}
                                                              onChange={handleStandardCheck}>
                                                        <ListItemIcon>
                                                            <Tooltip title={standard.description} placement={"bottom-start"}>
                                                            <FormControlLabel
                                                                control={
                                                                    <Checkbox
                                                                        icon={<StarBorder/>}
                                                                        checkedIcon={<Star/>}
                                                                        value={standard.id}
                                                                        checked={selectedStandards[standard.id] || false}/>}
                                                                label={standard.description}
                                                            />
                                                                </Tooltip>

                                                        </ListItemIcon>
                                                    </ListItem>
                                                </List>
                                            )
                                        })
                                        }
                                    </Collapse>
                                    </div>

                                </React.Fragment>
                            )
                        })}
                    </List>
                )
            })
        } else if (state.available_grouped_competencies && showCompetencyDrawer) {
            drawerContent = Array.from(state.available_grouped_competencies).map((course, index) => {
                return (
                    <List
                        key={`list-${index}`}
                        component="div"
                        aria-labelledby="nested-list-subheader"

                        subheader={
                            <ListSubheader
                                component="div"
                                inset={true}
                                classes={headerclasses}
                                className={headerclasses.root}
                                syles={{opacity: 1}}
                                id="nested-list-subheader">
                                {course.title}
                            </ListSubheader>
                        }
                        className={classes.root}
                    >
                        {course.groups.map((group, index) => {
                            return (
                                <React.Fragment key={`${group.id}-${index}`}>
                                    <ListItem key={index} button onClick={() => handleExpandTopic(index)}>
                                        <ListItemText primary={group.title}/>
                                        {expandSublist[index] ? <ExpandLess/> : <ExpandMore/>}
                                    </ListItem>
                                    <div style={{ width: '100%', overflow: "hidden", textOverflow: "ellipsis" }}>
                                    <Collapse in={expandSublist[index]} timeout="auto" unmountOnExit>
                                        {group.competencies.map((competency, index) => {
                                            const disabled = `${competency.id}` === `${assessmentCompetencyId}`
                                            return (
                                                <List key={`l-${index}`} component="div" disablePadding disabled={disabled}>
                                                    <ListItem
                                                        button
                                                        className={classes.nested}
                                                        disabled={disabled}
                                                        onChange={handleCompetencyCheck}
                                                    >
                                                        <Tooltip title={competency.description} placement={"bottom-start"}>
                                                        <ListItemIcon>
                                                            <FormControlLabel
                                                                control={
                                                                    <Checkbox
                                                                        icon={<TurnedInNot/>}
                                                                        checkedIcon={<TurnedIn/>}
                                                                        value={competency.id}
                                                                        disabled={disabled}
                                                                        checked={selectedCompetencies[competency.id.toString()] || false}/>}
                                                                label={competency.description}
                                                            />
                                                        </ListItemIcon>
                                                        </Tooltip>
                                                    </ListItem>
                                                </List>
                                            )
                                        })
                                        }
                                    </Collapse>
                                    </div>
                                </React.Fragment>
                            )
                        })}
                    </List>
                )
            })
        }

        return drawerContent
    }

    return (
    <React.Fragment>
        {loading ?
        <Paper  elevation={8} className={classes.root} id={'add-task-form'}>
            <Skeleton variant="rect" width={'100%'} height={500} />
        </Paper>
            :
        <Paper  elevation={8} className={classes.root} id={'add-task-form'}>
            {!editing ?  <Typography variant={'h6'}>{taskName}</Typography> :
            <FormControl className={''} fullWidth={true}>
                <TextField
                    value={taskName}
                    variant={'outlined'}
                    id={"task-name"}
                    onChange={handleTaskNameChange}
                    label={"Name"}
                    required
                />
            </FormControl> }
            <Grid container spacing={theme.spacing(.5)}>
                {standardRows()}
                {editing && <Grid item sm={12}>
                    <Tooltip title={!selectedCourses ? 'Select a course for the assessment in order ' +
                        'to be able to select standards for the assessment\'s tasks.  The courses used in the ' +
                        'selected section determine the standards available' : ''}>
                        <div>
                            <Button
                                variant="outlined"
                                color="secondary"
                                className={classes.button}
                                disabled={!selectedCourses}
                                onClick={handleOpenStandardsDrawer}>
                                Add / Remove Standards
                            </Button>
                        </div>
                    </Tooltip>
                </Grid>}
                {competencySection()}
                { editing && <Grid item sm={12}>
                    <Tooltip title={!selectedCourses ? 'Select a course for the assessment in order ' +
                        'to be able to select competencies. ' : ''}>
                        <div>
                            <Button
                                variant="outlined"
                                color="secondary"
                                className={classes.button}
                                disabled={!selectedCourses}
                                onClick={handleOpenCompetencyDrawer}>
                                Add / Remove Competencies
                            </Button>
                        </div>
                    </Tooltip>
                </Grid>
                }

                <Grid item sm={12}>
                    <Divider/>
                    { editing ?
                        <React.Fragment><Button onClick={handleUpdateTask}>Done</Button><Button onClick={handleDoneTask}>Cancel</Button></React.Fragment>
                    :
                        <React.Fragment>
                            <Button onClick={handleEdit}>Edit</Button>
                            { !isHolistic &&
                                <Tooltip
                                    title={task.deletable ? '' : 'Not deletable: Observations have been made on this task'}>
                                    <div>
                                        <Button onClick={doDelete} disabled={!task.deletable}>Delete</Button>
                                    </div>
                                </Tooltip>
                            }
                        </React.Fragment>
                    }
                </Grid>
            </Grid>
        </Paper>}
        <Drawer
            open={showStandardDrawer || showCompetencyDrawer}
            anchor={"right"}
            onClose={closeDrawer}
            classes={{
                paper: drawerClasses.drawerPaper
            }}
            className={drawerClasses.drawer}
        >
            <Button onClick={closeDrawer}>Done</Button>

            {getDrawerContents()}
        </Drawer>

    </React.Fragment>
    );
}
