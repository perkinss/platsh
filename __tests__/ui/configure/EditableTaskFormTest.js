import React from 'react'
import { createShallow, createRender } from '@material-ui/core/test-utils';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect'
import theme from "../../../app/javascript/styles/theme";
import {ThemeProvider } from '@material-ui/styles'
import {ConfigurationContext} from "../../../app/javascript/context/ConfigurationContext";
import EditableTaskForm from "../../../app/javascript/components/configure/EditableTaskForm";

Enzyme.configure({ adapter: new Adapter() });

describe('<EditableTaskForm />', () => {
    let shallow, render, container;
    let classes = {}
    let state = {
        available_content_standards: [{
            title: "Math 9",
            contents: [
                {   id: 1, name: "Know something", standards: [{ id: 3, description: "Add and subtract"}]}
            ]
        }],
        available_grouped_competencies: [
            {
                    groups: [
                        {
                            competencies: [
                                {id: 22, description: "Develop, demonstrate, and apply conceptual understand through play, story, inquiry, and problem solving"},
                                {id: 23, description: "Visualize to explore and illustrate mathematical concepts â€¨and relationships"}
                            ]
                        },
                    ]
            }]
    }

    let tasks = [{name: 'atask', competencies: [23,22], standards: {3: "M"}}]

    beforeEach(() => {
        shallow = createShallow();
        render = createRender();
        container = document.createElement('div');
        document.body.appendChild(container);
    });

    afterEach(() => {
        document.body.removeChild(container);
        container = null;
    })

    it('should match the snapshot', () => {
        const wrapper = shallow(
            <ThemeProvider theme={theme}>
                <ConfigurationContext.Provider value={{dispatch: () => {}, state: state }} >
                    <EditableTaskForm classes={classes} />
                </ConfigurationContext.Provider>
            </ThemeProvider>);
        expect(wrapper).not.toBeNull()
        expect(wrapper).toMatchSnapshot()
    });

    // todo: it would be nice to have where we can show that the form clears the assessemnt competency from the task
    it('should render the readonly form when the showForm is true', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                <ConfigurationContext.Provider value={{dispatch: () => {}, state: state }} >
                    <EditableTaskForm
                        classes={classes}
                        assessmentCompetencyId={'22'}
                        sectionID={1}
                        possiblyNewTask={tasks[0]}
                        editMode={false}
                    />
                </ConfigurationContext.Provider>
            </ThemeProvider>);
        let doc = wrapper.html()
        expect(doc).toContain('atask')
        expect(doc).toContain('Edit')
        expect(doc).toContain('Delete')
        expect(doc).not.toContain('Done')
        expect(doc).not.toContain('Cancel')
        expect(doc).toContain('Visualize to explore and illustrate')
    })

    it('should render the form with a task and its standards when there is a task', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                <ConfigurationContext.Provider value={{dispatch: () => {}, state: state }} >
                    <EditableTaskForm
                        classes={classes}
                        assessmentCompetencyId={''}
                        sectionID={1}
                        possiblyNewTask={tasks[0]}
                        editMode={false} />
                        }
                    />
                </ConfigurationContext.Provider>
            </ThemeProvider>);
        let doc = wrapper.html()
        expect(doc).toContain(tasks[0].name)
        expect(doc).toContain('>Standards:')
        expect(wrapper.text()).toContain('Add and subtract')
    })

    it('should be in edit mode when edit is true', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                <ConfigurationContext.Provider value={{dispatch: () => {}, state: state }} >
                    <EditableTaskForm
                        classes={classes}
                        assessmentCompetencyId={22}
                        sectionID={1}
                        possiblyNewTask={tasks[0]}
                        editMode={true} />
                    }
                    />
                </ConfigurationContext.Provider>
            </ThemeProvider>);
        let doc = wrapper.html()
        expect(doc).toContain(tasks[0].name)
        expect(doc).not.toContain('>Standards:')
        expect(wrapper.text()).toContain('Add and subtract')
        expect(wrapper.text()).toContain('Click to set the level of difficulty (Low, Medium or High):')
        expect(doc).toContain('Add / Remove Standards')
        expect(doc).toContain('Add / Remove Competencies')
        expect(doc).toContain('Done')
        expect(doc).toContain('Cancel')

    })


})