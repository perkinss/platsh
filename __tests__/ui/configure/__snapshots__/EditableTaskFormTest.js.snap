// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditableTaskForm /> should match the snapshot 1`] = `ShallowWrapper {}`;
